src/OpenTelemetry.Contrib.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -0,0 +1,104 @@
+            {
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}gc_heap_size", () => (double)(GC.GetTotalMemory(false) / 1_000_000), "MB", "GC Heap Size");
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}gen_0-gc_count", () => GC.CollectionCount(0), description: "Gen 0 GC Count");
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}gen_1-gc_count", () => GC.CollectionCount(1), description: "Gen 1 GC Count");
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}gen_2-gc_count", () => GC.CollectionCount(2), description: "Gen 2 GC Count");
+#if NETCOREAPP3_1_OR_GREATER
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}alloc_rate", () => GC.GetTotalAllocatedBytes(), "B", "Allocation Rate");
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}gc_fragmentation", GetFragmentation, description: "GC Fragmentation");
+#endif
+
+#if NET6_0_OR_GREATER
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}gc_committed", () => (double)(GC.GetGCMemoryInfo().TotalCommittedBytes / 1_000_000), "MB", description: "GC Committed Bytes");
+#endif
+            }
+
+#if NET6_0_OR_GREATER
+            if (options.IsJitEnabled)
+            {
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}il_bytes_jitted", () => System.Runtime.JitInfo.GetCompiledILBytes(), "B", description: "IL Bytes Jitted");
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}methods_jitted_count", () => System.Runtime.JitInfo.GetCompiledMethodCount(), description: "Number of Methods Jitted");
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}time_in_jit", () => System.Runtime.JitInfo.GetCompilationTime().TotalMilliseconds, "ms", description: "Time spent in JIT");
+            }
+#endif
+
+#if NETCOREAPP3_1_OR_GREATER
+            if (options.IsThreadingEnabled)
+            {
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}monitor_lock_contention_count", () => Monitor.LockContentionCount, description: "Monitor Lock Contention Count");
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}threadpool_thread_count", () => ThreadPool.ThreadCount, description: "ThreadPool Thread Count");
+                this.meter.CreateObservableGauge($"{options.MetricPrefix}threadpool_completed_items_count", () => ThreadPool.CompletedWorkItemCount, description: "ThreadPool Completed Work Item Count");
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}threadpool_queue_length", () => ThreadPool.PendingWorkItemCount, description: "ThreadPool Queue Length");
+                this.meter.CreateObservableCounter($"{options.MetricPrefix}active_timer_count", () => Timer.ActiveCount, description: "Number of Active Timers");
+            }
+#endif
+
+            if (options.IsPerformanceEnabled)
```

Comment:
Perhaps this should be named "Memory", or "Process Memory"? "Performance" seems very broad.

Reply:
This "category" is used to provide the cpu-usage as well, which was removed for the 1st version.
What about "ProcessPerformance"?

Reply:
I noticed later that we were planning to prefix all these metrics with "process.runtime.dotnet." and the OT semantic conventions suggest this value should be named something like "process.memory.xyz". Maybe we should leave this instrument out of this Meter and put it with a different collection of non-runtime process information?

Reply:
@noahfalk Does the [semantics ](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/runtime-environment-metrics.md#runtime-environment-specific-metrics---processruntimeenvironment) didn't say "process.runtime.xyz.memory.usage" ?

Reply:
Sorry I'm not sure I understood the question? 
I believe the semantics are distinguishing memory allocated by a runtime's GC (process.runtime.dotnet.something) vs. all memory allocated by the process for any purpose (process.memory.something). The original EventCounters for the runtime were created prior to these semantic conventions existing and they weren't attempting to separate process-wide information from runtime-specific information.

Reply:
Most of the OpenTelemetry semantic conventions are still at very early stages.
Take [process.memory.usage and process.memory.virtual](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/process-metrics.md#process) for example, the names are bit confusing, and memory measurements are much broader if we consider things like private working set, shared working set, commit size, reserved size, managed vs. native/unmanaged memory.

If I understand @noahfalk correctly, I think the intention is to put .NET runtime specific things under "process.runtime.dotnet.*", and put operating system specific things under a different place that doesn't have "runtime.dotnet" in the name. In other words, we need to ask ourselves - would this metric also apply to C++, if yes, it should not have "dotnet" in the name.